From 9bc1ec59bd8db07e41067717aeea2749314ec801 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 5 May 2022 12:16:57 +0100
Subject: [PATCH] configs: Enable DM_WRITECACHE module on BCM2711

For Pi 4 (which has high speed USB and PCIe), enable DM_WRITECACHE as a
module to allow an external SSD to be used as a fast write cache for a
slower storage device.

See: https://github.com/raspberrypi/linux/issues/5020

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/configs/bcm2711_defconfig   | 1 +
 arch/arm64/configs/bcm2711_defconfig | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/arm/configs/bcm2711_defconfig b/arch/arm/configs/bcm2711_defconfig
index 81b9b9803dcb..c6d09b1f6f1f 100644
--- a/arch/arm/configs/bcm2711_defconfig
+++ b/arch/arm/configs/bcm2711_defconfig
@@ -491,6 +491,7 @@ CONFIG_DM_CRYPT=m
 CONFIG_DM_SNAPSHOT=m
 CONFIG_DM_THIN_PROVISIONING=m
 CONFIG_DM_CACHE=m
+CONFIG_DM_WRITECACHE=m
 CONFIG_DM_MIRROR=m
 CONFIG_DM_LOG_USERSPACE=m
 CONFIG_DM_RAID=m
diff --git a/arch/arm64/configs/bcm2711_defconfig b/arch/arm64/configs/bcm2711_defconfig
index f763b3f666ae..1188bdc2a4b7 100644
--- a/arch/arm64/configs/bcm2711_defconfig
+++ b/arch/arm64/configs/bcm2711_defconfig
@@ -489,6 +489,7 @@ CONFIG_DM_CRYPT=m
 CONFIG_DM_SNAPSHOT=m
 CONFIG_DM_THIN_PROVISIONING=m
 CONFIG_DM_CACHE=m
+CONFIG_DM_WRITECACHE=m
 CONFIG_DM_MIRROR=m
 CONFIG_DM_LOG_USERSPACE=m
 CONFIG_DM_RAID=m
-- 
2.32.0

